microsetta_public_api/repo/_alpha_repo.py

```diff
@@ -1,11 +1,87 @@
+from microsetta_public_api import config
+from microsetta_public_api.exceptions import ConfigurationError, UnknownMetric
+from qiime2 import Artifact
+from q2_types.sample_data import AlphaDiversity, SampleData
+import os
+import pandas as pd
+
+
 class AlphaRepo:
 
+    # resources needs to be a class variable in order to be able to be
+    #  mocked in the test cases
+    resources = dict()
+    resource_locations = None
+
+    def __init__(self):
+        self.resource_locations = config.resources.get('alpha_resources', None)
+        if self.resource_locations is not None:
+            self._validate_resource_locations(self.resource_locations)
+            # leaving as None here allows lazy loading of resources
+            self.resources = {key: None for key in self.resource_locations}
+
+    @staticmethod
+    def _validate_resource_locations(resource_locations):
+        if not isinstance(resource_locations, dict):
+            raise ConfigurationError('`alpha_resources` must be '
+                                     'able to be parsed into a python '
+                                     'dictionary.')
+        all_keys_str = all(isinstance(key, str) for key in
+                           resource_locations)
+        if not all_keys_str:
+            raise ConfigurationError('All `alpha_resources` keys must be '
+                                     'strings.')
+        all_values_fp = all(os.path.exists(val) for val in
+                            resource_locations.values())
+        if not all_values_fp:
+            raise ConfigurationError('All `alpha_resources` values must be '
+                                     'existing file paths.')
+        return True
```

Comment:
this seems like it may be a better fit within a configuration manager which can assert the validity of a configuration file?

Reply:
https://github.com/biocore/microsetta-public-api/pull/6/commits/9f545b88117924801ce78b100adc1d71633f0505